packages/server-boilerplate/src/microService/api/ApiBuilder.ts

```diff
@@ -41,13 +44,23 @@ export class ApiBuilder {
     this.app.use(bodyParser.json({ limit: '50mb' }));
     this.app.use(errorHandler());
 
+    // micro-service connections
+    const cachedEntityServerConnection = new CachedOutboundConnection();
+
     /**
      * Add singletons to be attached to req for every route
      */
     this.app.use((req: Request, res: Response, next: NextFunction) => {
       req.models = this.models;
 
-      const context = {}; // context is shared between request and response
+      const microServiceAuthHandler = {
+        getAuthHeader: async () => req.headers.authorization || '',
+      };
+      const context: RequestContext = {
+        microServices: {
+          entityApi: new EntityApi(cachedEntityServerConnection, microServiceAuthHandler),
+        },
```

Comment:
We construct a single `cachedEntityServerConnection` for the whole server and then just wrap it in an `EntityApi` per request... Might play around a bit more with the composition of these classes as I think I can do a better job here...